dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/DataValueController.java

```diff
@@ -576,9 +576,16 @@ public void getDataValueFile(
         setNoStore( response );
         try
         {
-            InputStream file = fileResourceService.getFileResourceContent( fileResource );
-            response.setContentLength( file.available() );
-            IOUtils.copy( file, response.getOutputStream() );
+            long contentLength = fileResourceService.copyFileResourceContent( fileResource, response.getOutputStream() );
+
+            if ( contentLength <= Integer.MAX_VALUE )
```

Comment:
@vietnguyen there is a method `setContentLengthLong`  on ServletResponse / HttpServletResponse which you can use here instead.

https://docs.oracle.com/javaee/7/api/javax/servlet/ServletResponse.html#setContentLengthLong-long-

Reply:
fixed